packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -254,8 +256,20 @@ export class AuthUtil {
             throw new ToolkitError('Connection is not an SSO connection', { code: 'BadConnectionType' })
         }
 
-        const bearerToken = await this.conn.getToken()
-        return bearerToken.accessToken
+        try {
+            const bearerToken = await this.conn.getToken()
+            return bearerToken.accessToken
+        } catch (err) {
+            if (err instanceof ProfileNotFoundError) {
+                getLogger().info(
+                    'getBearerToken: logging out due to error encountered while fetching auth token: %s',
+                    err
+                )
+                await this.secondaryAuth.deleteConnection()
```

Comment:
Would this make sense to generalize for all Auths? In `Auth.getToken()` this try/catch would be there and when you delete it, it should emit `auth.onDidDeleteConnection()` and in SecondaryAuth we listen for that event which should trigger the change in AuthUtil. You can then for the focusQPanel call in AuthUtil have it trigger in `onDidChangeActiveConnection()` when the argument is undefined.

This is not a blocker to this PR, but please make a SIM if it makes sense to do this at a later point.